feat: import/export dashboards via cli

arpit-agarwal opened the pull request:
This PR is initial implementation to close #5821

@mistercrunch 
Please validate correctness of implementation

I will add documentation and test in couple of days.

Commit:
feat: import/export dashboards via cli

Commit:
style: fixed lint error

arpit-agarwal commented:
@mistercrunch I am not sure why `cypress` test is failing as there is no change related to that.
On my dev machine it is working fine. 
Please validate if any recent merge to master may causing this.

kristw commented:
That test is flaky. Please try to re-run. @graceguo-supercat and I are working to make it more stable. Sorry for the inconvenience.
On Sep 27, 2018, 23:42 -0700, Arpit <notifications@github.com>, wrote:
> @mistercrunch I am not sure why cypress test is failing as there is no change related to that.
> On my dev machine it is working fine.
> Please validate if any recent merge to master may causing this.
> @kristw This is related recent commit on master. can you help me in fixing cypress build here
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

arpit-agarwal commented:
Thanks for update @krstw.  This is open for review.  I will commit testcase and docstring docs in couple of days.

Commit:
test: added test for import and export util

Commit:
test: removing import test as it is causing integrity issues

Import is a wrapper around exist functionality so we can go ahead without a test or mock the actual db operation using https://docs.python.org/3/library/unittest.mock.html

And validate the wrapper operations only.

arpit-agarwal commented:
@mistercrunch @kristw 
I have added test for export cases. Import I have to revert as it was causing integrity error. 
I can mock the actual db import by mocking framework but decide against it as it is not used anywhere. 

The build is failing for unrelated test on async query celery. Can you help me fix that
```
test_run_async_query (tests.celery_tests.CeleryTestCase) ... 2018-10-01 11:35:56,304:INFO:root:The 'superset worker' command is deprecated. Please use the 'celery worker' command instead.
FAIL
```

Commit:
test: remove test data file

Commit:
test: removed usage of reserved keyword id

arpit-agarwal commented on tests/import_export_tests.py:
```diff
@@ -547,6 +550,34 @@ def test_import_druid_override_identical(self):
+
+        exported_dashboards = dashboards_objects['dashboards']
+        for dashboard in exported_dashboards:
+            id = dashboard.id
```
fixed

arpit-agarwal commented on tests/import_export_tests.py:
```diff
@@ -547,6 +550,34 @@ def test_import_druid_override_identical(self):
+
+        exported_tables = dashboards_objects['datasources']
+        for exported_table in exported_tables:
+            id = exported_table.id
```
fixed

michellethomas commented on tests/import_export_tests.py:
```diff
@@ -547,6 +550,34 @@ def test_import_druid_override_identical(self):
         self.assert_datasource_equals(
             copy_datasource, self.get_datasource(imported_id))
 
+    def test_export_dashboards_util(self):
```
@mistercrunch @arpit-agarwal I've noticed this test failing the py36-postgres tests when a pr has not made any changes to this code. For example https://travis-ci.org/apache/incubator-superset/jobs/436740957 Can you take a look?

williaster replied:
@mistercrunch @arpit-agarwal this is blocking merging all PRs into master

williaster replied:
@mistercrunch @arpit-agarwal this flaky test is still failing most builds for `py36-postgres` therefore blocking merges. Can you either PTAL or we will revert this PR if it's not fixed by noon today PST?

cc @john-bodley @kristw @michellethomas @graceguo-supercat

graceguo-supercat replied:
i made a test branch which reverted this whole PR, and it passed all CI tests. i will file PR to revert this. Thank you!

graceguo-supercat replied:
@arpit-agarwal Thank you very much for investigation. After I reverted this PR, what we observed is, most PRs passed all CI tests, includes postgres one. But master branch still failed at postgres test. Right now i am totally confused what exactly happened :(

arpit-agarwal replied:
As suggested we can look at the content of PR #5693  to fund the route cause.

I do see some changes in test in that PR related to test.
Any recent configruration changes on tarvis or postgress version or lsome python lib?

arpit-agarwal commented:
I quickliky looked at logs it seems this is the [first failure point](https://travis-ci.org/apache/incubator-superset/jobs/427942754) for PR #5880

arpit-agarwal commented:
@graceguo-supercat 

I see master build is passing now.  did we made some changes in configuration to fix it?

If all is going good can we raise The pR again to merge this feature.

If we are skeptical about test we can have a separate PR for test?

graceguo-supercat commented:
Yes i made some change from our travis setting. Sorry I reverted your PR and now think your PR should be good 👍. Do you want to create a new PR? i will approve it and merge.

arpit-agarwal commented:
Yes. I will create a new PT shortly

arpit-agarwal commented:
@graceguo-supercat  I have raised #6061 with these changes.

It will be great if we can document the travis changes for future reference.
